### PR TITLE
config: point `pa2024ch` to data prep with new weo 2023 scenarios

### PR DESCRIPTION
**Pull request by jdhoffa**
The newly created data directory now contains APS and STEPS scenarios for WEO 2023 for the aviation, cement and steel sectors. 

Importantly also, we had to update paths to the new (real) `peer` files, as those had to be re-run with those same latest scenarios. 

With these scenarios missing, the executive summary couldn't render correctly. 

Relates to:
RMI-PACTA/workflow.data.preparation/pull/238
RMI-PACTA/workflow.data.preparation/pull/237
RMI-PACTA/workflow.data.preparation/pull/236
RMI-PACTA/pacta.scenario.data.preparation/pull/61
RMI-PACTA/pacta.scenario.data.preparation/pull/63

### CODE REVIEW

**Comment by jdhoffa** [comment]
Open question to @AlexAxthelm and @cjyetman do you think I should update the `GENERAL` config as well? I don't think it is strictly necessary to do so... BUT I also can't think fo a strong reason NOT to.

It might make life less confusing down the road to keep both initiatives with the same data version.

**Comment by cjyetman** [comment]
I think that we have said that whatever happens with WEO will happen everywhere (PA2024CH, GENERAL, and all 2023Q4 initiatives), and I expect that word has traveled around, so I think we should follow through on that.

Additionally, until this issue is resolved, I prefer that we don't get into using different 2023Q4 datasets for different initiatives:
https://github.com/RMI-PACTA/workflow.data.preparation/issues/226

**Comment by jdhoffa** [comment]
Cool! done

**Review by cjyetman — APPROVED** [review]

**Comment by cjyetman** [comment]
presumably if #331 merges and this PR branch is updated, the tests will pass? so we're waiting on that, no?

**Comment by jdhoffa** [comment]
Indeed, depends on #331 

I will re-trigger CI/CD once that is merged.

**Comment by cjyetman** [comment]
🤔 
https://pactadatadev.blob.core.windows.net/ghactions-workflow-transition-monitor-results-reports/rmi_pacta_2023q4_pa2024ch-20240702T122952Z/EN/assetmanager/1/working_dir/50_Outputs/rmi_pacta_2023q4_pa2024ch/executive_summary/template.pdf
Screenshot 2024-07-02 at 15 22 32

**Comment by jdhoffa** [comment]
shit!

**Comment by jdhoffa** [comment]
Logs for this run can be found here:
https://portal.azure.com/#view/Microsoft_Azure_Storage/ContainerMenuBlade/~/overview/storageAccountId/%2Fsubscriptions%2Ffeef729b-4584-44af-a0f9-4827075512f9%2FresourceGroups%2FRMI-SP-PACTA-DEV%2Fproviders%2FMicrosoft.Storage%2FstorageAccounts%2Fpactadatadev/path/ghactions-workflow-transition-monitor-results-full/etag/%220x8DC3D63D9990D41%22/defaultEncryptionScope/%24account-encryption-key/denyEncryptionScopeOverride~/false/defaultId//publicAccessVal/None

**Comment by jdhoffa** [comment]
`error_messages.txt` was helpful XD
``` bash
2024-07-02 13:16:23.402782 ES: There was an error in prep/plot_scores(). Returning empty plot object.

2024-07-02 13:16:23.616256 ES: There was an error in prep/plot_scores(). Returning empty plot object.
```

**Comment by jdhoffa** [comment]
https://github.com/RMI-PACTA/pacta.executive.summary/blob/14575e1433e229154b84049d9a6b9c406e9d8366/inst/extdata/PA2024CH_en_exec_summary/scorecard.Rmd#L354

and

https://github.com/RMI-PACTA/pacta.executive.summary/blob/14575e1433e229154b84049d9a6b9c406e9d8366/inst/extdata/PA2024CH_en_exec_summary/scorecard.Rmd#L377

?

**Comment by jdhoffa** [comment]
@MonikaFu FYI, the problem seems to be that the `cement` sector isn't passing through properly.
When I run:
``` r
data_scores_b <- prep_scores(
      results_portfolio = results_portfolio,
      peers_results_aggregated = peers_results_aggregated,
      asset_class = "bonds",
      scenario_source = scenario_source
    )
```
I get:

Screenshot 2024-07-03 at 13 26 09

This seems to be what is causing the plot error.

**Comment by jdhoffa** [comment]
This is the error I get:
``` r
Error in `abort_if_invalid_values()` at pacta.executive.summary/R/plot_scores.R:227:3:
! Each value of `"sector"` must be one of these:
NA, power, automotive, coal, oil, gas, aviation, steel.
✖ You passed: cement.
Run `rlang::last_trace()` to see where the error occurred.
```

**Comment by jdhoffa** [comment]
Probably solved by https://github.com/RMI-PACTA/pacta.executive.summary/pull/348

**Comment by jdhoffa** [comment]
@AlexAxthelm @cjyetman @MonikaFu 
After some sleuthing locally, I have determined that the scorecard data still won't generate since the peer data does NOT contain the updated scenarios for the emission sectors. 

What that means we need to re-run the peer data (might already have been done by @AlexAxthelm , need to make sure the latest run contains STEPS, APS and NZE for the aviation, cement and steel sectors) and then either:

- [ ] Update the config to point to that new data OR
- [ ] Just update the existing testing peer data

This runs in parallel to a discussion we've already been having about updating our peer data in the CI/CD pipeline. 

Converting this PR to draft until the above has been completed.

**Comment by jdhoffa** [comment]
Latest bump of data_share_path now incorporates changes from https://github.com/RMI-PACTA/pacta.data.preparation/pull/32

**Comment by jdhoffa** [comment]
For the record: after a call with @AlexAxthelm we realized that we need the appropriate versions of `peer_results` and `user_data` also configured appropriately in this PR. 

To that end, I have asked him to target #333 with the #340 PR, and will merge it that together with this PR (I will update the PR title and description appropriately). 

I will also wait until @MonikaFu runs and updates the `user_data` (and get that on Azure` before re-running CI/CD here. 

After all the above is complete, we should have a good idea of if all IR and ES building runs well "in the wild". 
I would expect all plots to render correctly.

File: build/config/rmi_pacta_2023q4_general.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "data_share_path": "2023Q4_20240424T120055Z",
+  "data_share_path": "2023Q4_20240709T111731Z",
```
Comment (cjyetman) [inline build/config/rmi_pacta_2023q4_general.json:2]: side note: `data_share_path` is not a very intuitive name for the PACTA input directory

File: build/config/rmi_pacta_2023q4_pa2024ch.json [hunk 2]
```diff
@@ -1,12 +1,14 @@
 {
-  "data_share_path": "2023Q4_20240424T120055Z",
+  "data_share_path": "2023Q4_20240709T111731Z",
   "pacta_data_quarter": "2023Q4",
   "project_code": "PA2024CH",
   "templates_ref": "",
-  "peer_results": "https://pactadatadev.blob.core.windows.net/testing-files/peer-results/PA2024CH_peer-results-test",
+  "peer_results": "https://pactadatadev.blob.core.windows.net/project-files/pa2024ch/peer_files",
```
Comment (cjyetman) [inline build/config/rmi_pacta_2023q4_pa2024ch.json:6]: I know it's mentioned in the (very long) thread of this PR, but it's not super clear to me why/if the peer files should change here.

**Review by cjyetman — APPROVED** [review]
I can't / don't know how to review the test results anymore, but otherwise seems reasonable

**Comment by jdhoffa** [comment]
> az storage copy --recursive --source https://pactadatadev.blob.core.windows.net/ghactions-workflow-transition-monitor-results-reports-private/rmi_pacta_2023q4_pa2024ch-20240717T142912Z/EN/bank/2/working_dir/50_Outputs/rmi_pacta_2023q4_pa2024ch --destination rmi_pacta_2023q4_pa2024ch-EN-bank

To test you can run this command:
``` bash
az storage copy --recursive --source https://pactadatadev.blob.core.windows.net/ghactions-workflow-transition-monitor-results-reports-private/rmi_pacta_2023q4_pa2024ch-20240717T142912Z/EN/bank/2/working_dir/50_Outputs/rmi_pacta_2023q4_pa2024ch --destination rmi_pacta_2023q4_pa2024ch-EN-bank
```

However, since I believe you were never able to get the `az cli` up and running? It is also possible to access the files from the Azure GUI. 

From portal.azure.com, navigate to the `pactadatadev` > `Storage Browser` > `Blob containers` > `ghactions-workflow-transition-monitor-results-reports-private` and then find the file path as usual `rmi_pacta_2023q4_pa2024ch-20240717T142912Z/EN/bank/2/working_dir/50_Outputs/rmi_pacta_2023q4_pa2024ch --destination rmi_pacta_2023q4_pa2024ch-EN-bank`

I know it's not the most intuitive but definitely possible if you have the gumption 😉

File: build/config/rmi_pacta_2023q4_pa2024ch.json (hunk 2 above)
Comment (jdhoffa) [reply build/config/rmi_pacta_2023q4_pa2024ch.json:6]: Would you like me to document that in the PR body?

**Pull request merged**